lib/fluent/plugin/buffer.rb

```diff
@@ -352,6 +354,9 @@ def write(metadata_and_data, format: nil, size: nil, enqueue: false)
                 unstaged_chunks[metadata] ||= []
                 unstaged_chunks[metadata] << chunk
               end
+              if error and not error.empty?
```

Comment:
```diff
- not error.empty?
+ !error.empty?
```
is preferred in Fluend's code

Reply:
Yes, and it should be `if error && !error.empty?`, too.